### ISSUE TITLE
Como tratar quando o pedido não conseguiu ser enviado para o eSIC?

### ISSUE THREAD

**Issue by andresmrm** [okfn-brasil/cuidando2#23]
A pessoas enviou um pedido, o EsicLivre aceitou, mas na hora de repassar para o eSIC não foi. O que fazer?

**Comment by alexandre** [comment]
@andresmrm ,

A pagina não fala sobre validações (e.g. pedido repetido), então fica um pouco complicado. IMO, O que podemos fazer é uma validação básica no EsicLivre...e aguardar o eSIC. =/

**Comment by andresmrm** [comment]
Sim... A grande questão aí é que quando o usuário envia o pedido para nós, ele
não espera para ver se foi bem sucedido no eSIC. O pedido simplesmente é
colocado no nosso BD para envio e para o usuário está tudo certo. Porém,
quando o EsicLivre finalmente enviar o pedido no eSIC, pode dar erro. Mas
nesse momento o usuário pode nem estar mais navegando pelo site.

O que vejo de possibilidade aí seria enviar um e-mail para o usuário avisando
que o pedido dele deu pau, para tentar de novo se quiser...
Teria que marcar o pedido de algum jeito no BD também. Para que ele não
continue a tentar ser reenviado a cada 5s... Talvez tentar algumas vezes e
depois desistir marcando ele no BD como inválido e enviando um e-mail para o
usuário?

[2015-08-22 12:55] Alexandre:

**Comment by alexandre** [comment]
Eu acho que podemos considerar o "fail fast" e avisar no primeiro erro ao usuário que tem algum problema com o request (e.g. dados que ele passou.)

BTW, Eu acho que dá para considerar colocar um celery no projeto...para enviar esses emails e outras tarefas periodicas.

**Comment by andresmrm** [comment]
> Eu acho que podemos considerar o "fail fast" e avisar no primeiro erro ao
> usuário que tem algum problema com o request (e.g. dados que ele passou.)
> Não entendi... Nós vamos saber se deu erro no eSIC depois que o usuário não
> estiver mais esperando a resposta. Não dá para segurar o usuário até que o
> navegador do EsicLivre tenha tentado enviar o pedido dele, ainda mais porque
> podem ter vários na fila.
> 
> BTW, Eu acho que dá para considerar colocar um celery no projeto...para enviar
> esses emails e outras tarefas periodicas.
> É, também estou sentindo que sim...

**Comment by alexandre** [comment]
​Sim, por "fail fast" eu quero dizer enviar um e-mail assim que ocorrer o
primeiro erro ao invés de tentar outras vezes.

[ ]'s

**Comment by andresmrm** [comment]
Ah, entendi. =)
Dai vem a questão: como saber o e-mail do usuário? O EsicLivre acesso ao
e-mail da pessoa, pois ele não fica no token. Além disso, e-mail não é
obrigatório, pois a pessoa pode não ter e-mail no Facebook...

[2015-08-22 16:40] Alexandre:

**Comment by alexandre** [comment]
Email ou mensagem no Facebook? (dá para enviar mensagem via API?)

Entramos naquela questão do usuário receber notificações sobre o pedido;
essa seria uma (negativa, mas seria).

[ ]'s

2015-08-22 16:43 GMT-03:00 Andrés Martano notifications@github.com:

> Ah, entendi. =)
> Dai vem a questão: como saber o e-mail do usuário? O EsicLivre acesso ao
> e-mail da pessoa, pois ele não fica no token. Além disso, e-mail não é
> obrigatório, pois a pessoa pode não ter e-mail no Facebook...
> 
> [2015-08-22 16:40] Alexandre:
> 
> > ​Sim, por "fail fast" eu quero dizer enviar um e-mail assim que ocorrer o
> > primeiro erro ao invés de tentar outras vezes.
> > 
> > [ ]'s
> > 
> > —
> > Reply to this email directly or view it on GitHub.*
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/okfn-brasil/cuidando2/issues/23#issuecomment-133744920
> .

## 

## 

## Twitter: @_AleSou || GitHub: alexandre || Reddit: ubbersith

[...]o sentido disso tudo é que não há sentido em tentar enlouquecer
para impedir-se de ficar louco...guarde sua sanidade para mais
tarde[...] - Douglas Adams

**Comment by andresmrm** [comment]
Você quer dizer, enviar e-mail para quem tem e-mail e msg no Face para quem
logou via Face e sem e-mail? Meio complexo, não? Ainda mais se imaginarmos que
podemos adicionar login via outras coisas depois...
Talvez simplesmente não mandar e-mail para quem não tem e-mail? =P

[2015-08-22 16:49] Alexandre:

**Comment by alexandre** [comment]
Boa ideia, K.I.S.S.

2015-08-22 17:32 GMT-03:00 Andrés Martano notifications@github.com:

> Você quer dizer, enviar e-mail para quem tem e-mail e msg no Face para quem
> logou via Face e sem e-mail? Meio complexo, não? Ainda mais se imaginarmos
> que
> podemos adicionar login via outras coisas depois...
> Talvez simplesmente não mandar e-mail para quem não tem e-mail? =P
> 
> [2015-08-22 16:49] Alexandre:
> 
> > Email ou mensagem no Facebook? (dá para enviar mensagem via API?)
> > 
> > Entramos naquela questão do usuário receber notificações sobre o pedido;
> > essa seria uma (negativa, mas seria).
> > 
> > [ ]'s
> > 
> > 2015-08-22 16:43 GMT-03:00 Andrés Martano notifications@github.com:
> > 
> > > Ah, entendi. =)
> > > Dai vem a questão: como saber o e-mail do usuário? O EsicLivre acesso
> > > ao
> > > e-mail da pessoa, pois ele não fica no token. Além disso, e-mail não é
> > > obrigatório, pois a pessoa pode não ter e-mail no Facebook...
> > > 
> > > [2015-08-22 16:40] Alexandre:
> > > 
> > > > ​Sim, por "fail fast" eu quero dizer enviar um e-mail assim que
> > > > ocorrer o
> > > > primeiro erro ao invés de tentar outras vezes.
> > > > 
> > > > [ ]'s
> > > > 
> > > > —
> > > > Reply to this email directly or view it on GitHub.*
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> > > https://github.com/okfn-brasil/cuidando2/issues/23#issuecomment-133744920>
> > > .
> > 
> > ## 
> > 
> > ## 
> > 
> > ## Twitter: @_AleSou || GitHub: alexandre || Reddit: ubbersith
> > 
> > [...]o sentido disso tudo é que não há sentido em tentar enlouquecer
> > para impedir-se de ficar louco...guarde sua sanidade para mais
> > tarde[...] - Douglas Adams
> > 
> > —
> > Reply to this email directly or view it on GitHub.*
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/okfn-brasil/cuidando2/issues/23#issuecomment-133752188
> .

## 

## 

## Twitter: @_AleSou || GitHub: alexandre || Reddit: ubbersith

[...]o sentido disso tudo é que não há sentido em tentar enlouquecer
para impedir-se de ficar louco...guarde sua sanidade para mais
tarde[...] - Douglas Adams
